modules/documents/app/services/documents/oauth/ensure_application_service.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
+module Documents
+  module OAuth
+    class EnsureApplicationService < BaseServices::BaseCallable
+      APPLICATION_NAME = "Documents OAuth Application"
+
+      def perform
+        application = find_or_create_application
+
+        if application.persisted?
+          ServiceResult.success(result: application)
+        else
+          ServiceResult.failure(errors: application.errors)
+        end
+      end
+
+      private
+
+      def find_or_create_application
+        existing = Doorkeeper::Application.find_by(name: APPLICATION_NAME)
```

Comment:
I'd avoid searching/matching applications by name, because the name is supposed to be human-readable.

We could even think about localizing the name one day.

The machine readable identifier is the Client ID, which doorkeeper stores as `uid`. The `uid` is not required to be gibberish. It's perfectly allowed to be something like `hocuspocus`, so we can hardcode it as constant (like the name).